### DISCUSSION TITLE
Swappable argument serialization (eg `pickle`)

### DISCUSSION THREAD

**Discussion by RealOrangeOne** [RealOrangeOne/django-tasks#10] [Ideas]
Currently, arguments and return-values must be JSON serializable (using the default `json` encoder / decoder). This avoids a number of potential issues, at the detriment of flexibility.

Should this be something configurable? And if so, what are the most useful options (eg `pickle`, custom JSON encoder / decoder etc).

**Comment by zbyte64** [comment]
Would be nice to support pydantic models as well, though there's a difference between serialization (objects to primitives) and rendering (JSON or Pickle) so I am not sure if that should be handled separately.

**Comment by RealOrangeOne** [comment]
It's an intentional design decision to only support primitive types. Anything else will need a schema passed in a side-channel. Pickle would be too insecure.
